open-hackathon-server/src/hackathon/hack/team_manager.py

```diff
@@ -257,6 +298,14 @@ def update_team_member_status(self, operator, user_team_rel_id, status):
         if rel.user_id == team.leader_id:
             return precondition_failed("cannot update status of team leader")
 
+        # check if the user is already in this team.
+        members = self.db.find_all_objects_by(UserTeamRel, team_id=team.id, status=TeamMemberStatus.Approved)
+        if not members or len(members) == 0:
+            return not_found("this team doesn't have any members")
+        member_users_id = [m.user.id for m in members]
+        if rel.user_id in member_users_id:
+            return not_found("the user is already in this team.")
```

Comment:
What's the purpose of the above lines? I don't understand. Could u plz explain a little more?